--- a/fedot/core/operations/evaluation/operation_implementations/data_operations/ts_transformations.py
+++ b/fedot/core/operations/evaluation/operation_implementations/data_operations/ts_transformations.py
@@ -170,7 +170,7 @@
         all_transformed_features = None
         for current_ts_id in range(n_time_series):
             # For each time series in the array
-            if n_time_series == 1:
+            if len(features.shape) == 1:
                 current_ts = features
             else:
                 current_ts = np.ravel(features[:, current_ts_id])
